geomesa-native-api/src/main/java/org/locationtech/geomesa/api/GeoMesaIndex.java

```diff
@@ -0,0 +1,35 @@
+package org.locationtech.geomesa.api;
+
+import com.vividsolutions.jts.geom.Geometry;
+
+import java.util.Date;
+
+/**
+ * GeoMesaIndex is an API for utilizing GeoMesa as a spatial index
+ * without bringing in the Geotools SimpleFeature data model
+ * @param <T>
+ */
+public interface GeoMesaIndex<T> {
+
+    /**
+     * Query a GeoMesa index
+     * @param query
+     * @return
+     */
+    Iterable<T> query(GeoMesaQuery query);
+
+    /**
+     * Put a value in the GeoMesa index
+     * @param t
+     * @param geometry
+     * @param dtg
+     */
+    void put(T t, Geometry geometry, Date dtg);
```

Comment:
if we only use the Z3 index it's a little rough to actually do updates and deletes without knowing the geometry beforehand